lib/typescript/components/message/MessageInfoWrapper/index.tsx

```diff
@@ -12,23 +13,67 @@ type MessageInfoWrapperProps = {
   sentAt?: string;
   decoration?: ReactNode;
   senderName?: string;
+  deliveryState?: string;
+  messageId?: string;
+  resendFailedMessage?: (resend: boolean, messageId: string) => void;
 };
 
 export const MessageInfoWrapper = (props: MessageInfoWrapperProps) => {
-  const {sentAt, contact, fromContact, children, lastInGroup, isChatPlugin, decoration, senderName} = props;
+  const {
+    sentAt,
+    contact,
+    fromContact,
+    children,
+    lastInGroup,
+    isChatPlugin,
+    decoration,
+    senderName,
+    deliveryState,
+    messageId,
+    resendFailedMessage,
+  } = props;
 
   const isContact = isChatPlugin ? !fromContact : fromContact;
   const senderIdentity = sentAt ? ` - sent by ${senderName}` : `sent by ${senderName}`;
 
+  const handleResendFailedMessage = () => {
+    resendFailedMessage(true, messageId);
```

Comment:
The name of this prop is confusing: The function doesn't actually resend the message but marks it as to be resend (?)

Also the retry is in the wrong place. I'd suggest that the class only exposes a `onRetryMessage` that gets passed to the `onClick` handler